Geoip1 should still be available even when Geoip2 is available

mattab opened the pull request:
Because geoip1 can still be useful even when geoip2 is available (for example, geoip1 seems faster)

(this is a patch we hot-fixed on Cloud)

cc @sgiehl

sgiehl commented:
I would strongly recommend **NOT** to merge this. It was fully on purpose that it is not possible to switch to GeoIP Legacy once GeoIP2 is active. Switching from GeoIp2 back to GeoIp Legacy will disrupt the region data already tracked with GeoIp2. The conversion script included is able to convert legacy (FIPS) data to ISO. A conversion in the other direction is not possible. And having data in the wrong order (first ISO then FIPS) in the database will result in weird regions when running the conversion script.

Btw. disabling the GeoIp2 plugin should also "reactivate" the Legacy providers. Which should imho be enough. Also the integrated auto update / wizard will only be able to download GeoIp2 databases as long as the plugin is active.

tsteur commented:
Ideally we don't always have to patch it 👍

sgiehl committed:
Move provider visibility check to UI only
That should allow setting and using invisible geolocation providers without problems

sgiehl commented:
@tsteur @mattab  I've update the PR and moved the visibility check for providers from "backend" to "UI" only. That way the providers won't be shown in the UI (unless they are active) and can't be choosen easily. But they can still be chosen using the API, as they are still "available" in the list of all providers

mattab commented:
Good idea, this should solve the issue @sgiehl
